### PR TITLE
tree selection (#241)

### PR DESCRIPTION
**Pull request by FredLL-Avaiga**
- node expansion is different from node selection
- children_only_selection allows to disable selection of non leaf nodes
- row_height allows to specif a height for each row to homogeneize the rows with or without Icon

### CODE REVIEW

**Review by dinhlongviolin1 — APPROVED** [review]
Looks good to me

File: gui/doc/tree.csv [hunk 1]
```diff
@@ -1,4 +1,6 @@
 ﻿Name,Type,Default,Doc
 expanded,dynamic(bool|str[]),True,"if boolean value False: only one node can be expanded at one given level else it should be an array of the ids that need to be expanded"
 multiple,bool,FALSE,"If True, the user can select multiple items by holding the Ctrl key while clicking on items."
+children_only_selection,bool,False,"If True, the user can only select leaf nodes"
```
Comment (FabienLelaquais) [inline gui/doc/tree.csv]: select_leafs_only?
enable_non_leaf_selection?
all_nodes_selectable?
only_leafs_selectable?

**Review by FabienLelaquais — APPROVED** [review]
Well done.

File: gui/doc/tree.csv (hunk 1 above)
Comment (FredLL-Avaiga) [reply gui/doc/tree.csv]: I see that you can't find a good one either :-D

Comment (FredLL-Avaiga) [reply gui/doc/tree.csv]: I'll go for `select_leafs_only`

**Pull request merged**